lib/Audio/StreamGenerator.pm

```diff
@@ -4,11 +4,18 @@ our $VERSION = 0.06;
 
 use strict;
 use warnings;
-use Log::Log4perl qw(:easy);
+use Carp;
 
-Log::Log4perl->easy_init($DEBUG);
-
-my $logger = Log::Log4perl->get_logger('Audio::StreamGenerator');
+sub debug {
+    my ($self, $message) = @_;
+    return unless $self->{debug};
+    if (ref $self->{debug} eq 'CODE') {
+        &{ $self->{debug} }($message);
```

Comment:
Probably better written as `$self->{debug}->($message);` (left-to-right dereferencing)